<?xml version="1.0"?>
<launch>

	<arg name="namespace" default="/cam3d"/>
	<arg name="rate" default="5.0"/>
  	<arg name="start_manager" default="true"/>
  	<arg name="nodelet_manager" default="cam3d_throttle_nodelet_manager"/>

  	<node if="$(arg start_manager)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)"  args="manager" output="screen"/>

	<node pkg="nodelet" type="nodelet" name="cam3d_throttle"  args="load cob_cam3d_throttle/Cam3DThrottle $(arg nodelet_manager)" output="screen">
		<remap from="rgb_cloud_in" to="$(arg namespace)/depth_registered/points"/>
		<remap from="rgb_image_in" to="$(arg namespace)/rgb/image_color"/>
		<remap from="rgb_caminfo_in" to="$(arg namespace)/rgb/camera_info"/>
		<remap from="depth_image_in" to="$(arg namespace)/depth/image_raw"/>
		<remap from="cloud_in" to="$(arg namespace)/depth/points"/>
		<remap from="rgb_cloud_out" to="$(arg namespace)_throttled/depth_registered/points"/>
		<remap from="rgb_image_out" to="$(arg namespace)_throttled/rgb/image_color"/>
		<remap from="rgb_caminfo_out" to="$(arg namespace)_throttled/rgb/camera_info"/>
		<remap from="depth_image_out" to="$(arg namespace)_throttled/depth/image_raw"/>
		<remap from="cloud_out" to="$(arg namespace)_throttled/depth/points"/>
		<param name="max_rate" value="$(arg rate)"/>
	</node>
</launch>
